docs/site/LB3-vs-LB4-request-response-cycle.md

```diff
@@ -0,0 +1,448 @@
+through the sequence. It identifies the responsible handler for the requested
+endpoint and passes on the request to the handler.
+
+![LoopBack 4 request/response components](./imgs/lb4-req-res.png)
+
+Unlike LoopBack 3, models in LoopBack 4 do not create REST endpoints. Use the
+`lb4 controller` command to quickly generate the REST endpoints for a model.
+
+For more details, refer to the LoopBack 4
+[request/response cycle](https://loopback.io/doc/en/lb4/Request-response-cycle.html)
+doc.
+
+#### Access to the request/response object
+
+Since LoopBack 3 uses the Express middleware pattern, the request and response
+objects can always be accessed in all the middleware functions.
+
+In LoopBack 4, the request and response objects can be accessed in
+routers loaded using the
+[app.mountExpressRouter() ](https://loopback.io/doc/en/lb4/apidocs.rest.restapplication.mountexpressrouter.html)
+method, using the familiar Express middleware signature.
+
+Controllers, services, and repositories are LoopBack 4 components that
+participate in the request/response cycle. The request and response objects can
+be made available to them via
+[dependency injection](https://loopback.io/doc/en/lb4/Dependency-injection.html)
+.
+
+Example of accesssing the request and response object in a Controller:
+
+```ts
+import {inject} from '@loopback/context';
+import {Request, Response, RestBindings, get} from '@loopback/rest';
+
+export class ExampleController {
+  constructor(
+    @inject(RestBindings.Http.REQUEST) private req: Request,
+    @inject(RestBindings.Http.RESPONSE) private res: Response
+    ) {}
+
+  @get('/headers')
+  headers() {
+    // Sends back the request headers
+    this.res.send(this.req.headers);
+  }
+}
+```
+
+Similarly, the request and response objects can be injected into services
+and respositories along with other objects from the
+[context](https://loopback.io/doc/en/lb4/Context.html).
+
+{% include tip.html content="It may be tempting to use an Express router
+(because of familiarity) instead of a controller to add custom endpoints to the
+app, but bear it in mind that controllers are way more powerful and capable than
+Express routers because of their support for dependency injection and access to
+the application context." %}
+
+#### Data validation
+
+LoopBack 3 models come with
+[validation methods](https://loopback.io/doc/en/lb3/Validating-model-data.html)
+. The validation rules are derived from the model definition files or are
+applied by calling the validation methods in the model's JavaScript file.
+
+In LoopBack 4, models are defined as classes whose properties decorated with the
+[@property()](https://loopback.io/doc/en/lb4/Model.html#property-decorator)
+decorator become the model's properties. Request body to an endpoint is
+validated against the attributes specified in the `@property()` decorator of the
+corresponding model.
+
+Model validation methods like `validatesLengthOf()`,
```

Comment:
We now have the validation docs explaining the 3 layers data validation in lb4, see https://github.com/strongloop/loopback-next/blob/master/docs/site/Validation.md (maybe not published yet)
You probably want to add it as a reference here :p